Sources/SwiftFormat/PrettyPrint/PrettyPrint.swift

```diff
@@ -569,6 +584,40 @@ public class PrettyPrinter {
         write(",")
         spaceRemaining -= 1
       }
+
+    case .enableFormatting(let enabledPosition):
+      // if we're not disabled, we ignore the token
+      if let disabledPosition {
+        let start = source.utf8.index(source.utf8.startIndex, offsetBy: disabledPosition.utf8Offset)
+        let end: String.Index
+        if let enabledPosition {
+          end = source.utf8.index(source.utf8.startIndex, offsetBy: enabledPosition.utf8Offset)
+        } else {
+          end = source.endIndex
+        }
+        var text = String(source[start..<end])
+        // strip trailing whitespace so that the next formatting can add the right amount
+        if let nonWhitespace = text.rangeOfCharacter(
+          from: CharacterSet.whitespaces.inverted, options: .backwards) {
+          text = String(text[..<nonWhitespace.upperBound])
+        }
```

Comment:
How does the printer pick up the whitespace characters that were dropped here? Ie. how does it know to format those but not anything in `text`? Seems like I’m missing something here.

Reply:
Well, we have to make sure our state variables are set correctly so that it does that correctly. That the next few lines. 🙂

Reply:
I’m still missing how the whitespace characters that we’re shaving off with this are being printed.

If you have the following

```swift
func foo()   ⏩{}⏪
```

Then I think this only adds `func foo()` to the output stream, right (no whitespace)? But the next token that we print is the `{`, which doesn’t have any leading trivia (spaces are trailing trivia to `)`).

If those examples work because of some trivia re-attribution rule, how about the following?

```swift
func foo() /**/  ⏩{}⏪
```

```swift
func foo() /**/
  ⏩{}⏪
```